source/core_http_client.c

```diff
@@ -553,6 +570,29 @@ static uint32_t getZeroTimestampMs( void )
 
 /*-----------------------------------------------------------*/
 
+static int8_t caseInsensitiveStringCmp( const char * str1,
+                                        const char * str2,
+                                        size_t n )
+{
+    assert( str1 != NULL );
+    assert( str2 != NULL );
+
+    for( ; n > 0; n--, str1++, str2++ )
+    {
+        assert( str1 != NULL );
+        assert( str2 != NULL );
+
+        if( ( toupper( *str1 ) != toupper( *str2 ) ) )
+        {
+            return 1;
+        }
+    }
+
+    return 0;
```

Comment:
Some things:
* `n--`: MISRA doesn't like it when the values of the input arguments are altered
* `n--, str1++, str2++`: I'm like 80% sure MISRA doesn't like it when there are multiple updates in the `for`.
* `return 1;`: MISRA doesn't like it when there are multiple returns in a function
* `assert( str1 != NULL ); assert( str2 != NULL );`: I'm pretty sure the second set is redundant since this will only happen when either address is > `UINT_MAX - n`. If we trust that the size that's passed in is accurate then this can't happen; otherwise, we'd be accessing out of bounds anyway.

Consider an alternate implementation to appease MISRA, e.g.:
```suggestion
    size_t i = 0U;

    assert( str1 != NULL );
    assert( str2 != NULL );

    for( i = 0U; i < n; i++ )
    {
        if( toupper( str1[ i ] ) != toupper( str2[ i ] ) )
        {
            break;
        }
    }

    return ( i == n ) ? 0 : 1;
```